attributes.go

```diff
@@ -86,29 +86,29 @@ func addAddressAttributes(attrs []attribute.KeyValue, address string) []attribut
 	return append(attrs, semconv.NetPeerNameKey.String(address))
 }
 
-func statusCodeAttribute(protocol string, serverErr error) (attribute.KeyValue, bool) {
-	// Following the respective specifications, use integers and "status_code" for
-	// gRPC codes in contrast to strings and "error_code" for Connect codes.
-	switch protocol {
-	case grpcProtocol, grpcwebProtocol:
-		codeKey := attribute.Key("rpc." + protocol + ".status_code")
-		if serverErr != nil {
-			return codeKey.Int64(int64(connect.CodeOf(serverErr))), true
-		}
-		return codeKey.Int64(0), true // gRPC uses 0 for success
-	case connectProtocol:
+func statusCodeAttributes(onStreaming bool, serverErr error) []attribute.KeyValue {
+	var attributes []attribute.KeyValue
+
+	if serverErr != nil {
```

Comment:
We've carefully removed allocations from the core paths in the interceptor. To avoid them here, we should refactor this method to be similar to `addAddressAttributes` above where it takes a slice and appends to it, returning the modified slice.

Then we need to update all callers to ensure they've presized their slices with enough capacity to account for an additional attribute here.

As part of this, we should do a https://pkg.go.dev/golang.org/x/perf/cmd/benchstat run that verifies we aren't allocating more than before.

Reply:
@pkwarren I have updated the method to `addStatusCodeAttributes`. I have pasted the `benchstat` result on the PR description. The number of allocations remains the same after the changes.